src/server/server.py

```diff
@@ -0,0 +1,139 @@
+# coding=utf-8
+# Copyright 2018-2020 EVA
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import asyncio
+import string
+import os
+
+from signal import signal
+from signal import SIGINT, SIGTERM, SIGHUP, SIGUSR1
+
+from src.server.networking_utils import realtime_server_status,\
+    set_socket_io_timeouts
+
+from src.utils.logging_manager import LoggingManager, LoggingLevel
+
+from src.server.command_handler import handle_request
+
+
+class EvaServer(asyncio.Protocol):
+
+    """
+    Receives messages and offloads them to another task for processing them.
+
+    - It never creates any asynchronous tasks itself
+    - It doesn't have to know anything about any event loops
+    - It tracks its progress via the class-level counters
+    """
+
+    # These counters are used for realtime server monitoring
+    __connections__ = 0
+    __errors__ = 0
+
+    def connection_made(self, transport):
+        self.transport = transport
+
+        # Set timeout for sockets
+        if not set_socket_io_timeouts(self.transport, 60, 0):
```

Comment:
Shouldn't the timeouts be configurable? 

Maybe, we can use them as attributes in EvaServer? 
eg. EvaServer(timeout_secs=...). By doing this we can read the config only when EvaServer in instantiated.

Reply:
Yes, added it to the eva.yml file.